types.go

```diff
@@ -50,7 +50,10 @@ type Scope interface {
 	Histogram(name string, buckets Buckets) Histogram
 
 	// Tagged returns a new child scope with the given tags and current tags.
-	Tagged(tags map[string]string) Scope
+	Tagged(tags map[string]string, tenancys ...string) Scope
+
+	// TaggedWithTenancy required the context to extract the tenancy value, return a new child scope with tenancy tag plus given tags
+	TaggedWithTenancy(tenancy string, tags map[string]string) Scope
```

Comment:
this technically is a backwards incompatible change:
https://blog.merovius.de/2015/07/29/backwards-compatibility-in-go.html
see "Adding a method to an interface"

which will require a new major release.

The functionality could be implemented without changing the interface (e.g., `tally.WithTenancy(scope tally.Scope, tenancy... string) tally.Scope` which would avoid a breaking change.

Another simpler option is just to have users add the tag themselves, and export a tenancy tag name.

Reply:
Fair point, although I'm not yet sure if we're going to observe the struct-fields-could-break-embedded-structs concern. Regardless - I think we should keep tenancy rooted, so that it's not something that suddenly appears somewhere in the chain of scopes at runtime - in other words, I like the idea of creating a subscope with tenancy less than I like creating the root scope with a tenancy in the first place. If you're going to use tenancy with your scope, it should be included all the way down. Thus, maybe `NewRootScopeWithTenancy(ScopeOptions, time.Duration, string)`?